src/ledger/LedgerManagerImpl.cpp

```diff
@@ -1360,13 +1363,47 @@ LedgerManagerImpl::processFeesSeqNums(
         auto header = ltx.loadHeader().current();
         std::map<AccountID, SequenceNumber> accToMaxSeq;
 
+        // If we have expected results, we assign them to the mutable tx results
+        // here.
+        std::optional<std::vector<TransactionResultPair>::const_iterator>
+            expectedResultsIter = std::nullopt;
+        if (expectedResults)
```

Comment:
I think we should add an invariant either here or in TransactionFrame that makes sure we're actually doing catchup. What do you think?

Reply:
Is it possible for `CatchupManagerImpl::catchupWorkIsDone()` to be true along with a non null `mCatchupWork` here? I don't recall how the catchup state transitions work. Maybe @marta-lokhova has some input on a safe way to do this check.

Reply:
Since this change targets a very specific use case (parallel catchup for testing), I'd recommend putting all this functionality behind BUILD_TESTS. The prod paths should remain unchanged.

Reply:
Ah I thought this was meant for running catchup in general. Is it just for parallel catchup testing?

Reply:
Yes, we don't win much by skipping work in prod paths. Different application flows increase the risk of divergence. Parallel catchup seems like a nice middle ground given the performance benefits + low risk.

Reply:
I agree. In that case, a lot more of this PR should be behind the BUILD_TESTS ifdef.